### ISSUE TITLE
./configure deadlock

### ISSUE THREAD

**Issue by dschwoerer** [boutproject/BOUT-dev#744]
I tried to run configure on archer, but instead of failing, it called itself.
The configure command run is then:
```
  $ ./configure script found. Specify path using --with-netcdf option 5937 5
```
I guess the offending line is:
```
       [AC_MSG_ERROR([No configure script found. Specify path using --with-netcdf option])],
```
which results in:
```
test -n "$NCCONF" || NCCONF="as_fn_error $? "No configure script found. Specify path using --with-netcdf option" "$LINENO" 5"
```

Not quite sure why the configure script calls it self ...

**Comment by ZedThree** [comment]
That's very odd. Which branch is this on? And can you post the output of `module list`?

**Comment by dschwoerer** [comment]
Note, if I set the paths for linking and headers, but don't set PATH to include nc-config, it still fails on archer ...

A work around is to make sure nc-config is available ...

This was on my_v4, but the same issue is on next.
```
Currently Loaded Modulefiles:
  1) modules/3.2.10.6                      10) packages-archer                       19) python-compute/2.7.6                  28) gni-headers/4.0-1.0502.10859.7.8.ari
  2) eswrap/1.3.3-1.020200.1278.0          11) bolt/0.6                              20) pc-numpy/1.9.2-libsci                 29) xpmem/0.1-2.0502.64982.5.3.ari
  3) switch/1.0-1.0502.60522.1.61.ari      12) nano/2.2.6                            21) pc-scipy/0.15.1-libsci                30) dvs/2.5_0.9.0-1.0502.2188.1.116.ari
  4) gcc/6.3.0                             13) leave_time/1.3.0                      22) pc-netcdf4-python/1.1.5-python2       31) alps/5.2.4-2.0502.9774.31.11.ari
  5) craype-network-aries                  14) quickstart/1.0                        23) cray-libsci/16.11.1                   32) rca/1.0.0-2.0502.60530.1.62.ari
  6) craype-ivybridge                      15) ack/2.14                              24) udreg/2.3.2-1.0502.10518.2.17.ari     33) atp/2.1.0
  7) craype/2.5.10                         16) xalt/0.6.0                            25) ugni/6.0-1.0502.10863.8.29.ari        34) PrgEnv-gnu/5.2.82
  8) pbs/12.2.401.141761                   17) epcc-tools/6.0                        26) pmi/5.0.12
  9) cray-mpich/7.5.5                      18) fftw/3.3.4.11                         27) dmapp/7.0.1-1.0502.11080.8.76.ari
```
But even if netcdf is not available, it should fail with an error, instead of being stuck

**Comment by ZedThree** [comment]
Should be fixed in #747 -- can you check please?

There's an additional bug in finding HDF5 on Archer as well, because they don't have `h5cc`, which we need to check the settings... I'll have to find some alternative means for getting the flags on Archer.
